src/PreferencesMatrixGenerator.java

```diff
@@ -0,0 +1,210 @@
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.util.Random;
+
+class PreferencesMatrixGenerator
+{
+    enum PreferenceGeneratorMode
+    {
+        kBestCase,
+        kWorstCase,
+        kRandom;
+    }
+
+    // Usage: PreferencesMatrixGenerator NumberOfPreferences Mode "OutputFile.txt"
+    // Example: PreferencesMatrixGenerator 10 0 "test1.txt"
+    public static void main(String args[])
+    {
+        final int kExpectedNumberOfArguments = 3;
+
+        if (kExpectedNumberOfArguments == args.length)
+        {
+            final int kDataSetSize = Integer.parseInt(args[0]);
+            final int kMode = Integer.parseInt(args[1]);
+            final String kFileName = args[2];
+
+            PreferenceGeneratorMode mode = PreferenceGeneratorMode.kBestCase;
+
+            switch(kMode)
+            {
+                case 0:
+                    mode = PreferenceGeneratorMode.kBestCase;
+                    break;
+                case 1:
+                    mode = PreferenceGeneratorMode.kWorstCase;
+                    break;
+                case 2:
+                    mode = PreferenceGeneratorMode.kRandom;
+                    break;
+                default:
+                    System.out.println("Invalid Mode!");
+                    System.out.println("Valid Modes: BestCase:0 WorstCase:1 Random:2");
+                    System.exit(-100);
+            }
+
+            PreferencesMatrixGenerator preferencesMatrixGenerator = new PreferencesMatrixGenerator(kDataSetSize, mode);
+            //preferencesMatrixGenerator.printItems();
+            preferencesMatrixGenerator.writeToFile(kFileName);
+        }
+        else
+        {
+            System.out.println("Invalid Number of Arguments");
+            System.out.println("Usage: PreferencesMatrixGenerator NumberOfPreferences Mode \"outputFile.txt\"");
+            System.out.println("Example: PreferencesMatrixGenerator 100 0 \"test1.txt\"");
+            System.out.println("Valid Modes: BestCase:0 WorstCase:1 Random:2");
+            System.exit(-100);
+        }
+    }
+
+    private int m_dataSize;
+    private int[][] m_menPreferenceMatrix;
+    private int[][] m_womenPreferenceMatrix;
+    private PreferenceGeneratorMode m_mode;
+
+    PreferencesMatrixGenerator(int dataSize, PreferenceGeneratorMode mode)
+    {
+        m_dataSize = dataSize;
+        m_mode = mode;
+        m_menPreferenceMatrix = new int[m_dataSize][m_dataSize];
+        m_womenPreferenceMatrix = new int[m_dataSize][m_dataSize];
+
+        populatePreferenceMatrices();
+    }
+
+    private void populatePreferenceMatrices()
+    {
+        switch (m_mode)
+        {
+
+            case kBestCase:
+                // Best case is when men and women have the same preference lists
```

Comment:
Isn't the best case when each pair is each other's first option?
Let's say in 3x3 preference matrix
```
1 2 3
2 3 1
3 1 2

1 2 3
2 3 1
3 1 2
```
This way man 1 proposes to first woman and she accepts and man 1 never gets rejected because woman 1 never gets another proposal because she is not the first option for anybody else.

Reply:
Yes, I will fix this.